### ISSUE TITLE
cmake-built executable doesn't set fish directories correctly

### ISSUE THREAD

**Issue by mqudsi** [fish-shell/fish-shell#4825]
It seems that the binary built via cmake has the install path baked in. Here are two consecutive runs of fish `master`, one built via autotools and the other via cmake:

```fish
mqudsi@Blitzkrieg /m/d/r/fish> ./fish # built via autotools
Welcome to fish, the friendly interactive shell
mqudsi@Blitzkrieg /m/d/r/fish> echo $fish_function_path
/home/mqudsi/.config/fish/functions /mnt/d/rand/fish/etc/functions /usr/local/share/fish/vendor_functions.d /mnt/d/rand/fish/share/functions
mqudsi@Blitzkrieg /m/d/r/fish> :
mqudsi@Blitzkrieg /m/d/r/fish> exit
mqudsi@Blitzkrieg /m/d/r/fish> build/fish # built with cmake
Welcome to fish, the friendly interactive shell
mqudsi@Blitzkrieg /m/d/r/fish> echo $fish_function_path
/home/mqudsi/.config/fish/functions /usr/local/etc/fish/functions share/fish/vendor_functions.d /usr/local/share/fish/functions
```

`PWD` was the `fish` src directory at `/mnt/d/rand/fish/src/`

This means that changes to completions, functions, etc. in the fish source code aren't picked up by the cmake-built fish until `make install` is run.

**Comment by ridiculousfish** [comment]
fish tries to be relocatable, which means that the directory tree can be moved and fish will continue working. fish currently looks for the ./share and ./etc directories adjacent to fish, to support the autotools build. Since the ./etc directory is not critical for running fish, I'm going to relax this to just the ./share directory, which allows the cmake build directory to be recognized as a "relocatable" directory.

**Comment by ridiculousfish** [comment]
I reverted my fix because it was bogus. Anyways this isn't about compiled-in paths (which are the same for cmake and autotools) but rather the way in which fish identifies if it is in a relocatable directory. It only affects users who build fish and run it from the build directory, and so I don't think it has to be fixed in 3.0.

**Comment by mqudsi** [comment]
Thanks, @ridiculousfish. So installing to `/usr` instead of `/usr/local` will run into no problems?

**Comment by ridiculousfish** [comment]
If you wish to install to `/usr` one can set `CMAKE_INSTALL_PREFIX` to `/usr`, similarly to how you might run `./configure --preifx=/usr`.

**Comment by mqudsi** [comment]
Yup, but I wasn't sure if this PR was a symptom of code not respecting that value (although I don't think that's the case).

**Comment by mqudsi** [comment]
The core issue persists in that fish does not pick up on being run from the out-of-tree build dir and looks for functions, scripts, etc. in /usr/local/etc instead of in the source or build directories.

**Comment by faho** [comment]
Fixed by 6bc59db7213a756460f948ad92455403edab715b.

**Comment by mqudsi** [comment]
Thanks @faho
